lib/engine/route.rb

```diff
@@ -19,6 +19,7 @@ def initialize(game, phase, train, **opts)
       @revenue_str = opts[:revenue_str]
       @subsidy = opts[:subsidy]
       @halts = opts[:halts]
+      @abilities = opts[:abilities] || []
```

Comment:
by setting [] here you will always store an empty array in the database because array is truthy maybe this should be nil or in the Action::Route code you should remove empty abilities

Reply:
Have refactored it a bit now.